### DISCUSSION TITLE
Duplicate Row action fail to set {{ Action 1.Duplicated row._id }} when passed to a open screen "modal"

### DISCUSSION THREAD

**Discussion by matteolavaggi** [Budibase/budibase#12780] [Help]
Hi, i have a problem with an action performed inside a button click.

The required function is very simple:

I have a form populated with the data of a row, in different screens, in some it is a form and in others a side panel opened directly from the row. In these forms I have several buttons, one of which is "edit" which loads the _id value of the row into a state variable (row_id) and then opens an edit screen that loads and filters a datasource using the value present in row_id. The edit button works, clicking on edit opens the editing form in "modal" mode with the data of the correct row.

Now in the same forms I have a second "duplicate" button that performs these actions:
Duplicate Row
Reads the _id of the duplicate row from the duplicate row action, saves this _id in the row_id state variable and calls the edit screen again to modify the newly duplicated row.

The problem is that the row is DUPLICATED but the edit form opens with the default data of a row which is not the duplicated one and is not even the one previously clicked (the one FROM which we duplicated). The same thing happens if I add a DUPLICATE button inside the edit form itself and try to duplicate the row currently being edited and then reload the edit form with the _id of the row just duplicated.

This is the "duplicate" button that does not work.
<img width="842" alt="image" src="https://github.com/Budibase/budibase/assets/24798703/9c91858e-e021-4acd-8842-981269d6d79a">
<img width="991" alt="image" src="https://github.com/Budibase/budibase/assets/24798703/84e9e36d-8efc-4eb7-a58d-4194ea571667">
<img width="979" alt="image" src="https://github.com/Budibase/budibase/assets/24798703/440dec41-bf09-420a-ba02-bdf50047ae50">

This is the "edit" button that work
<img width="977" alt="image" src="https://github.com/Budibase/budibase/assets/24798703/ecece00d-0d66-4f5e-8eeb-26a1be1b2c81">
<img width="973" alt="image" src="https://github.com/Budibase/budibase/assets/24798703/d7735b01-6cfc-4fe5-a3d0-4da671832df8">

They do the same thing but the "duplicate" open the edit screen with a default row data not the duplicated one.

Edit on domusbet row open edit screen with domusbet data:
<img width="429" alt="image" src="https://github.com/Budibase/budibase/assets/24798703/c89431ba-3650-4b1d-9918-59960055964e">

Duplicate on domusbet duplicate the row but after that open the edit screen with a default data (the one i can see when edit the screen):
<img width="399" alt="image" src="https://github.com/Budibase/budibase/assets/24798703/f3a8f473-a76f-4836-859f-f30c70d37bba">

**Comment by melohagan** [comment]
Hey @matteolavaggi 

There appears to be a misunderstanding on the usage of app state here. When using the **Navigate To** action for edit forms, you will want to use [URL variables](https://docs.budibase.com/docs/url-parameters) and pass the ID in the URL, and not use app state.

e.g. `/bet-edit-form/{{ Details form block.Data Entry._id }}`

You can then filter your 'bet-edit-form' with the binding: `{{ URL.id }}` assuming you have configured the screen route to take the format: `/bet-edit-form/:id`
